### PR TITLE
IWB-3: added waitlist functionality

### PR DESCRIPTION
**Pull request by FrankreedX**
Again, just like IWB-2, this is meant to be the base implementation, not polished

### CODE REVIEW

File: app/segments/(team)/chooseTeam.js [hunk 1]
```diff
@@ -64,7 +92,7 @@ function ChooseTeam() {
         }}
         refreshControl={<RefreshInvalidate invalidateKeys={invalidateKeys} />}
```
Comment (Gehrkej) [inline app/segments/(team)/chooseTeam.js:93]: Don't know if it was PR but when I refresh this screen after requesting to join the team, there is an animation that takes me to the same page. Might be worth looking into more.

File: dbOperations/hooks/useWaitlist.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { db } from "~/firebaseConfig";
+
+export const useWaitlist = ({ enabled = true } = {}) => {
+  const { currentTeamId } = useAuthContext();
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["waitlist"],
+    queryFn: async () => {
+      console.log("fetching waitlist");
+      const newWaitlist = {};
+      // Fetch all time records
```
Comment (Gehrkej) [inline dbOperations/hooks/useWaitlist.js:13]: looks like a stray comment from a ctrl C + ctrl V lol
```suggestion
      // Fetch all Entries on the Waitlist
```

File: dbOperations/addToWaitlist.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  try {
+    const newRequestRef = doc(
+      db,
+      "teams",
+      currentTeamId,
+      "waitlist",
+      currentUserId,
+    );
+    const currentTime = Date.now();
+    console.log("data", currentUserInfo);
```
Comment (Gehrkej) [inline dbOperations/addToWaitlist.js:18]: Might not want this log when we merge it
```suggestion
```

File: app/content/team/index.js [hunk 4]
```diff
@@ -144,6 +147,11 @@ function Index() {
         return priority1 - priority2;
       }
 
+      //doesn't softlock in case displayName is null for some reason
+      if (!user1.name) {
+        return -1;
+      }
+
```
Comment (Gehrkej) [inline app/content/team/index.js:154]: Is this just a hotfix for the search functionality on the team page?

**Review by Gehrkej — CHANGES_REQUESTED** [review]
Overall looks good and seems to function properly. One question, a little refresh bug and a little bit of code clean up but great job 😀

File: app/content/team/index.js (hunk 4 above)
Comment (FrankreedX) [reply app/content/team/index.js:154]: The logic right now is pretty solid so this if is not tripped, but while I was working on this PR and messed with displayName, I managed to soft-lock myself a lot of times, so I decided to leave it.

Comment (Gehrkej) [reply app/content/team/index.js:154]: Fair, should we keep it in just in case then?

Comment (FrankreedX) [reply app/content/team/index.js:154]: ye

File: app/segments/(team)/chooseTeam.js (hunk 1 above)
Comment (FrankreedX) [reply app/segments/(team)/chooseTeam.js:93]: I think this is in IWB-4 and I've just pushed a fix. Can you check it out again?

**Pull request merged**